framework/src/executioners/FEProblemSolve.C

```diff
@@ -404,7 +405,16 @@ FEProblemSolve::solve()
         if (is_nonlinear)
           _problem.solve(sys->number());
         else
-          _problem.solveLinearSystem(sys->number() - _problem.numNonlinearSystems());
+        {
+          const unsigned int linear_sys_number = sys->number() - _problem.numNonlinearSystems();
+          _problem.solveLinearSystem(linear_sys_number, &_problem.getPetscOptions());
+
+          // This is for postprocessing purposes in case none of the objects
```

Comment:
so if it s for postprcessing purpose, does it need to be performed on every multi-system iteration or could it be performed at the end of the loop ?